pkgs/shells/zsh/nix-zsh-completions/default.nix

```diff
@@ -15,15 +15,20 @@ stdenv.mkDerivation rec {
   };
 
   installPhase = ''
-    mkdir -p $out/share/zsh/site-functions
+    runHook preInstall
+
+    mkdir -p $out/share/zsh/{site-functions,plugins/nix}
     cp _* $out/share/zsh/site-functions
+    cp *.zsh $out/share/zsh/plugins/nix
+
+    runHook postInstall
```

Comment:
As previously mentioned, don't run these hooks manually here (for both derivations).